packages/autocomplete-js/src/__tests__/autocomplete.test.ts

```diff
@@ -151,6 +155,90 @@ describe('autocomplete-js', () => {
     `);
   });
 
+  test("renders with an auto-incremented id if there's multiple instances", async () => {
+    const container = document.createElement('div');
+
+    document.body.appendChild(container);
+    autocomplete({
+      container,
+    });
+    autocomplete({
+      container,
+    });
+    autocomplete({
+      container,
+    });
+
+    await waitFor(() => {
+      expect(
+        document.querySelector('#autocomplete-0-label')
+      ).toBeInTheDocument();
+      expect(
+        document.querySelector('#autocomplete-1-label')
+      ).toBeInTheDocument();
+      expect(
+        document.querySelector('#autocomplete-2-label')
+      ).toBeInTheDocument();
+    });
+  });
+
+  test("doesn't increment the id when toggling detached mode", async () => {
+    const container = document.createElement('div');
+
+    document.body.appendChild(container);
+    const { update } = autocomplete<{ label: string }>({
+      container,
+    });
+
+    await waitFor(() => {
+      expect(
+        document.querySelector('.aa-DetachedSearchButton')
+      ).not.toBeInTheDocument();
+      expect(
+        document.querySelector('#autocomplete-3-label')
+      ).toBeInTheDocument();
+    });
+
+    Object.defineProperty(window, 'matchMedia', {
+      writable: true,
+      value: jest.fn((query) => ({
+        matches: true,
+        media: query,
+        onchange: null,
+        addListener: jest.fn(),
+        removeListener: jest.fn(),
+        addEventListener: jest.fn(),
+        removeEventListener: jest.fn(),
+        dispatchEvent: jest.fn(),
+      })),
+    });
+
+    update({ detachedMediaQuery: '' });
+
+    Object.defineProperty(window, 'matchMedia', {
```

Comment:
You can store the previous value before overriding it to restore it.

Reply:
Fixed in https://github.com/algolia/autocomplete/pull/489/commits/1fc7f0bb36cb67008d035295cdcb7e7c40e3bbeb